// socket-api-package1/api/socket-connection.js
// Tests the 'Socket IO' API loader

exports.sockets = [
    {
        event: 'admin.task',
        onEvent: ({}, callback) => {
            callback(null, 'admin task complete');
        },
        accessLevel: 'admin'
    },
    {
        event: 'staff.task',
        onEvent: ({}, callback) => {
            callback(null, 'staff task complete');
        },
        accessLevel: 'staff'
    },
    {
        event: 'user.task',
        onEvent: ({}, callback) => {
            callback(null, 'user task complete');
        },
        accessLevel: 'user'
    },
    {
        event: 'public.task',
        onEvent: ({}, callback) => {
            callback(null, 'public task complete');
        }
    },
    {
        event: 'read.books',
        httpMethod: 'GET',
        onEvent: ({}, callback) => {
            callback(null, 'got books');
        },
        scope: [
            'read:books'
        ]
    },
    {
        event: 'write.books',
        httpMethod: 'POST',
        onEvent: ({}, callback) => {
            callback(null, 'wrote book');
        },
        scope: [
            'write:books'
        ]
    }
]
