// socket-api-package1/api/socket-notifications.js
'use strict';

// This represents a persistent database for notifications
let pretendNotificationsStore = {
    user: [],
    staff: [
        'Archived notification 1'
    ],
    admin: [
        'Archived admin notification 1'
    ]
}, staffRoom = 'staff:notifications';

function processSomething(callback) {
    setTimeout(() => {
        callback();
    }, 0);
}

exports.sockets = [
    {
        event: 'subscribe:notifications',
        onEvent: ({socket, message}, callback) => {
            const {username} = socket.user;

            // Place the socket in a user-specific channel for notifications (a.k.a. "room")
            socket.join(`${username}:notifications`, () => {
                callback(null, pretendNotificationsStore[username]);
            });
        },
        middleware: [],
        accessLevel: 'user'
    },
    {
        event: 'process:something',
        onEvent: ({socket}, callback) => {
            const {username} = socket.user;

            // Do processing
            processSomething(() => {
                let notification = `Something was processed by ${username}!`;

                socket.broadcast.to(`${username}:notifications`).emit(`${username}:notification`, notification);
                pretendNotificationsStore[username].push(notification);

                callback(null, 'processing finished');
            });
        },
        accessLevel: 'user'
    }
];
