// api-package1/api/api1Service.js
function api1Service() {

    function userTask(req, res) {
        res.send('user task complete');
    }

    function staffTask(req, res) {
        res.send('staff task complete');
    }

    function publicTask(req, res) {
        res.send('public task complete');
    }

    function adminTask(req, res) {
        res.send('admin task complete');
    }

    function getBooks(req, res) {
        res.send('got books');
    }

    function writeBook(req, res) {
        res.send('wrote book');
    }

    return {
        routes: [
            {path: '/user/task', httpMethod: 'GET', middleware: userTask, accessLevel: 'user'},
            {path: '/public/task', httpMethod: 'GET', middleware: publicTask},
            {path: '/staff/task', httpMethod: 'GET', middleware: staffTask, accessLevel: 'staff'},
            {path: '/admin/task', httpMethod: 'GET', middleware: adminTask, accessLevel: 'admin'},
            {
                path: '/books',
                httpMethod: 'GET',
                middleware: getBooks,
                scope: [
                    'read:books'
                ]
            },
            {
                path: '/books',
                httpMethod: 'POST',
                middleware: writeBook,
                scope: [
                    'write:books'
                ]
            },
            {
                path: '/books/empty/scope',
                httpMethod: 'GET',
                middleware: getBooks,
                scope: []
            }
        ]
    };
}

module.exports = api1Service;
